pkg/controllers/chart/chart.go

```diff
@@ -269,7 +289,21 @@ func (c *Controller) OnRemove(key string, chart *v1.HelmChart) (*v1.HelmChart, e
 		chartCopy.Status.JobName = job.Name
 		newChart, err := c.helms.Update(chartCopy)
 		if err != nil {
-			return chart, fmt.Errorf("unable to update status of helm chart to add uninstall job name %s", chartCopy.Status.JobName)
+			// if chart is gone, clean resources
+			if apierrors.IsNotFound(err) || strings.Contains(err.Error(), "StorageError") {
```

Comment:
I think you can use https://pkg.go.dev/k8s.io/apiserver/pkg/storage#IsInvalidObj

```suggestion
			if apierrors.IsNotFound(err) || storage.IsInvalidObj(err) {
```

Reply:
I tried, it is not like normal k8s package for client usage.
```
apierrors "k8s.io/apimachinery/pkg/api/errors"
```

Reply:
I don't see anything that prevents you from importing `"k8s.io/apiserver/pkg/storage"`?

Reply:
Wait my validation, it may not work as expected when using this package.

The log shows it does not go to the expected if, and the job is still left.

![image](https://github.com/k3s-io/helm-controller/assets/31133476/27431ce9-bb91-4f81-89de-4322316f9078)
```
harv41:/home/rancher # kk get jobs -A
NAMESPACE               NAME                               COMPLETIONS   DURATION   AGE
cattle-logging-system   helm-delete-rancher-logging        1/1           4s         4m4s
```

Reply:
`storage.IsInvalidObj` is not working as expected, revert first.